src/pages/index.vue

```diff
@@ -1096,7 +1101,26 @@ export default {
             query: { q: this.searchQuery },
           })
         );
+      } else {
+        const fallbackKeyword = this.randomKeywords?.[0];
+        this.$router.push(
+          this.localeLocation({
+            name: 'store',
+            query: { q: fallbackKeyword },
+          })
+        );
+      }
+    },
+    getRandomPlaceholder() {
+      if (this.$i18n.locale === 'zh-Hant') {
+        const shuffled = [...SEARCH_SUGGESTIONS].sort(
+          () => Math.random() - 0.5
+        );
+        const randomTerms = shuffled.slice(0, 3);
+        this.randomKeywords = randomTerms;
+        return randomTerms.join('、');
       }
+      return this.$t('gutenberg_search_placeholder');
     },
```

Comment:
Can be refactored to a searchMixin?

Reply:
Since the two use cases(homepage/ bookstore) are different, refactor it as a mixin would require more modifications. so I think it doesn't need to be made into a mixin.